### PR TITLE
Project Movies - Michael & Nabeel

### PR DESCRIPTION
**Pull request by michaelchangdk**
Link to my deployed project:

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/Pages/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+// Import URLS
+import { URLS } from "utils/urls";
+
+const Home = () => {
+
+    const [isLoading, setIsLoading] = useState(true)
+    const [popular, setPopular] = useState([])
+
+    useEffect(() => {
```
Comment (SteppbySteph) [inline code/src/Pages/Home.jsx:16]: We are really impressed about your code of how to fetch the API here using 'async', 'Promise.all()'. Well done!

File: code/src/Pages/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                .then((res) => res.json())
+            ));
+            setIsLoading(false)
+            setPopular(results => [...result1.results, ...result2.results])
+        }
+
+        fetchMovies();
+    }, [])
+
+    return(
```
Comment (SteppbySteph) [inline code/src/Pages/Home.jsx:30]: Such a clean and readable code structure. :)

File: code/src/Pages/Home.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                <Route
+                    path="*"
+                    element={<Error/>}
+                />
+            </Routes>
+        </BrowserRouter>
+    )
+}
+
+export default Home
```
Comment (SteppbySteph) [inline code/src/Pages/Home.jsx:58]: You can add another empty row here to get rid of the circle symbol.

File: code/src/Pages/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }, [])
+
+    return(
+        <BrowserRouter>
+            <Routes>
+                <Route 
+                    path="/" 
+                    element={
+                    <>
+                    {isLoading && <LoadingSpinner />}
```
Comment (SteppbySteph) [inline code/src/Pages/Home.jsx:37]: Nice to see that you guys doing a styled component for the LoadingSpinner. You really have embraced the knowledge about creating and using this technique.

File: code/src/Components/LoadingSpinner.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+const LoadingSpinner = () => {
+    return (
+        <Loader />
+    )
+}
+
+export default LoadingSpinner
+
+const Loader = styled.div`
```
Comment (SteppbySteph) [inline code/src/Components/LoadingSpinner.jsx:12]: Such a long and impressive styling of your LoadingSpinner! Great job with the result!

File: code/src/Components/FilmList.jsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        </Link>
+        )
+        })}
+        </Main>
+    )
+}
+
+export default FilmList
+
+// Styled Components
```
Comment (SteppbySteph) [inline code/src/Components/FilmList.jsx:30]: You did a wonderful job putting together the different stylings in this component. We learn a lot reading this page and how you guys apply the styling in your return code!! Standing ovation!

File: code/src/Components/FilmList.jsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  padding: 20px;
+`
+
+const FilmImageGroup = styled.div`
+  position: relative;
+  display: flex;
+  width: 100%;
+  height: 100%;
+
+  &:hover ${FilmImageOverlay} {
```
Comment (SteppbySteph) [inline code/src/Components/FilmList.jsx:80]: Interesting way of adding style with javascript. Curious to know if you had to google a long time to find this solution.

File: code/src/Components/Details.jsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+const Details = ({films}) => {
+    const {id} = useParams()
+    const movie = films.filter(movie => movie.id === Number(id))[0];
+
+    const backgroundImageUrl = `https://image.tmdb.org/t/p/original${movie?.backdrop_path}`
+    
+    if (movie) {
+    return(
+        // <section style={backgroundStyle}>
+        <Section backgroundimage={backgroundImageUrl}>
```
Comment (SteppbySteph) [inline code/src/Components/Details.jsx:17]: We like how you approached this solution of adding a URL to a styled component tag. The logic seem so obvious and once again gives us a 'Aha'- moment.

File: code/src/Components/Error.jsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import error from '../assets/error.png'
+import { Link } from "react-router-dom";
+import styled from 'styled-components'
+
+const Error = () => {
+    return(
+        <ErrorDiv>
+            <Link to="/">
+                <ErrorImage src={error} alt="circle with x inside" />
```
Comment (SteppbySteph) [inline code/src/Components/Error.jsx:10]: Nice styling with the error symbol!

File: code/src/Pages/Home.jsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
+import React, {useState, useEffect} from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import LoadingSpinner from "Components/LoadingSpinner";
```
Comment (SteppbySteph) [inline code/src/Pages/Home.jsx:3]: Here you can add space between dependencies you download and the components you import according to Technigo's convention.

**Review by toal13 — COMMENTED** [review]
Great job Michael and Nabeel!
There are so many good features in your project. It was fun to see the code and we learned a lot from code reviewing  your project. The only thing to remember is to add extra row after your last type to get rid of the red symbol in github.

Stephannie and Tomoyo

**Pull request closed without merging**